File: simple-network-monitor-frontend/src/app/entities/sample.ts
import { SampleType } from "./sampleType";
import { Sensor } from "./sensor";
import { Status } from "./status";
import { Page } from "./page";

export interface SampleHal {
    _embedded: SampleEmbedded;
    page : Page;
}

export interface SampleEmbedded {
    samples : Sample[];
}
export interface Sample {
    id: string;
    type: SampleType;
    time: Date;
    value: number;
    warn: number;
    critical: number;
    min: number;
    max: number;
    unit: string;
    msg: string;
    status: Status;
    sensor: Sensor;
}

File: simple-network-monitor-backend/src/main/java/io/github/johannesschaefer/simplenetworkmonitor/entities/Sample.java
package io.github.johannesschaefer.simplenetworkmonitor.entities;

import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.Date;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Sample {
    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "org.hibernate.id.UUIDGenerator")
    private String id;
    @NonNull
    @ManyToOne(optional = false)
    private SampleType type;
    @NonNull
    @Builder.Default
    @Column(nullable = false)
    private Date time = new Date();
    private Double value;
    private Double warn;
    private Double critical;
    private Double min;
    private Double max;
    private String unit;
    private String msg;
    @NonNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;
    @NonNull
    @ManyToOne(optional = false)
    private Sensor sensor;
}
